### PR TITLE
Follow up features and fixes for message validation - part 1

### PR DESCRIPTION
**Pull request by thalesmg**
Continuation of #12711 .

Fixes https://emqx.atlassian.net/browse/EMQX-12054

Release version: e5.7

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx/src/emqx_hookpoints.erl [hunk 1]
```diff
@@ -59,6 +59,7 @@
     'message.publish',
     'message.puback',
     'message.dropped',
+    'message.validation_failure',
```
Comment (zmstone) [inline apps/emqx/src/emqx_hookpoints.erl:62]: maybe beter to follow the majority verb style naming.
```suggestion
    'message.validation_failed',
```

File: apps/emqx_rule_engine/src/emqx_rule_events.erl [hunk 2]
```diff
@@ -93,6 +95,7 @@ event_topics_enum() ->
         '$events/message_delivered',
         '$events/message_acked',
         '$events/message_dropped',
+        '$events/message_validation_failure',
```
Comment (zmstone) [inline apps/emqx_rule_engine/src/emqx_rule_events.erl:98]:
```suggestion
        '$events/message_validation_failed',
```

File: apps/emqx_rule_engine/src/emqx_rule_events.erl [hunk 3]
```diff
@@ -221,6 +224,19 @@ on_message_dropped(Message, _, Reason, Conf) ->
     end,
     {ok, Message}.
 
+on_message_validation_failure(Message, ValidationContext, Conf) ->
+    case ignore_sys_message(Message) of
+        true ->
+            ok;
+        false ->
+            apply_event(
+                'message.validation_failure',
```
Comment (zmstone) [inline apps/emqx_rule_engine/src/emqx_rule_events.erl:233]:
```suggestion
                'message.validation_failed',
```

File: apps/emqx_rule_engine/src/emqx_rule_events.erl [hunk 4]
```diff
@@ -477,6 +493,38 @@ eventmsg_dropped(
...
+        topic = Topic,
+        headers = Headers,
+        payload = Payload,
+        timestamp = Timestamp
+    },
+    ValidationContext
+) ->
+    #{name := ValidationName} = ValidationContext,
+    with_basic_columns(
+        'message.validation_failure',
```
Comment (zmstone) [inline apps/emqx_rule_engine/src/emqx_rule_events.erl:511]:
```suggestion
        'message.validation_failed',
```

File: apps/emqx_rule_engine/src/emqx_rule_events.erl [hunk 5]
```diff
@@ -627,6 +675,7 @@ event_info() ->
         event_info_message_deliver(),
         event_info_message_acked(),
         event_info_message_dropped(),
+        event_info_message_validation_failure(),
```
Comment (zmstone) [inline apps/emqx_rule_engine/src/emqx_rule_events.erl:678]:
```suggestion
        event_info_message_validation_failed(),
```

File: apps/emqx/src/emqx_hookpoints.erl (hunk 1 above)
Comment (thalesmg) [reply apps/emqx/src/emqx_hookpoints.erl:62]: I followed the EIP: https://github.com/emqx/eip/blob/c898b13394160f0694e75a3fc5e7db9768377aa3/active/0024-message-validation.md?plain=1#L122

**Review by zmstone — APPROVED** [review]

**Comment by emqxqa** [comment]
[Test Execution]()

**Pull request merged**